java-checks/src/main/java/org/sonar/java/checks/IgnoredReturnValueCheck.java

```diff
@@ -67,14 +68,15 @@ public class IgnoredReturnValueCheck extends IssuableSubscriptionVisitor {
     "java.time.ZonedDateTime",
     "java.math.BigInteger",
     "java.math.BigDecimal",
-    "java.util.Optional",
+    JAVA_UTIL_OPTIONAL,
     "com.google.common.base.Optional");
 
   private static final List<String> EXCLUDED_PREFIX = Arrays.asList("parse", "format", "decode", "valueOf");
 
   private static final MethodMatchers EXCLUDED = MethodMatchers.or(
     MethodMatchers.create().ofTypes("java.lang.Character").names("toChars").addParametersMatcher("int", "char[]", "int").build(),
-    MethodMatchers.create().ofTypes(JAVA_LANG_STRING).names("intern").addWithoutParametersMatcher().build());
+    MethodMatchers.create().ofTypes(JAVA_LANG_STRING).names("intern").addWithoutParametersMatcher().build(),
```

Comment:
What is this `intern` method and why do we need to support it?

Reply:
intern, make a copy of the string and store it in the constant pool; this can be used to improve performance when you use it a lot of times the same string.

Reply:
I don't think intern is used a lot, but when it is, it is for the side effect

Reply:
Why do we need to add it to the list? It looks like it is [already ignored in the code sample](https://github.com/SonarSource/sonar-java/blob/3e2cb4478100a5ea082dde30004067579eef58a4/java-checks-test-sources/default/src/main/java/checks/S2201_IgnoredReturnValueCheckSample.java#L77). So this new line does not seem to have any effect on the existing code

Reply:
I did add it, it was here before. The line has changed because I added a new case at the end of the or matcher (we don't see it with this view). The only logical change is at line 79.